CONTRIBUTING.md

```diff
@@ -0,0 +1,150 @@
+
+In general, you should try to test a single feature in one spec file. Try to not have too many or too few tests in one file. However, there is no golden rule about that.
+
+One spec file represents a scenario to test a behavior from a user point of view.
+
+Example of scenario:
+
+1. Load the `examples/e-commerce/` page
+2. Click on "Appliances" category
+3. Click on rating "4 & up"
+4. Check if the result list matches the expected one
+
+Which can be translated in a spec file to:
+
+```js
+describe('InstantSearch - Search on specific brand and query filtering', () => {
+  it('navigates to the e-commerce demo', async () => {
+    await browser.url('examples/e-commerce/');
+  });
+
+  it('selects "Appliances" category', async () => {
+    await browser.setSelectedHierarchicalMenuItem('Appliances');
+  });
+
+  it('selects "4 & up" rating', async () => {
+    await browser.setRatingMenuValue('4 & up');
+  });
+
+  it('must have the expected results', async () => {
+    const hitsTitles = await browser.getHitsTitles();
+
+    expect(hitsTitles).toEqual([
+      'Nest - Learning Thermostat - 3rd Generation - Stainless Steel',
+      'LG - 1.1 Cu. Ft. Mid-Size Microwave - Stainless-Steel',
+      'Insignia™ - 2.6 Cu. Ft. Compact Refrigerator - Black',
+      'Keurig - K50 Coffeemaker - Black',
+      'iRobot - Roomba 650 Vacuuming Robot - Black',
+      'Shark - Navigator Lift-Away Deluxe Bagless Upright Vacuum - Blue',
+      'LG - 1.5 Cu. Ft. Mid-Size Microwave - Stainless Steel',
+      'Insignia™ - 5.0 Cu. Ft. Chest Freezer - White',
+      'Samsung - activewash 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
+      'Samsung - 4.8 Cu. Ft. 11-Cycle High-Efficiency Top-Loading Washer - White',
+      'LG - 2.0 Cu. Ft. Full-Size Microwave - Stainless Steel',
+      'Shark - Rotator Professional Lift-Away HEPA Bagless 2-in-1 Upright Vacuum - Red',
+      'LG - 4.5 Cu. Ft. 8-Cycle High-Efficiency Top-Loading Washer - White',
+      'Samsung - 4.2 Cu. Ft. 9-Cycle High-Efficiency Steam Front-Loading Washer - Platinum',
+      'Panasonic - 1.3 Cu. Ft. Mid-Size Microwave - Stainless steel/black/silver',
+      'LG - 2.0 Cu. Ft. Mid-Size Microwave - Black Stainless',
+    ]);
+  });
+});
+```
+
+General guidelines when writing tests:
+
+- Each step should be translated as a `it` function, while this is not mandatory it helps to make the scenario more readable and properly [separate each actions in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).
+- All actions on the browser are asynchronous, so be sure to always `await` them. **Never run multiple asynchronous commands in parallel as it can confuse some browsers (Internet Explorer)**.
```

Comment:
Is using await really necessary in our case? Since we're using the bakedin [test runner](https://github.com/algolia/instantsearch-e2e-tests/blob/master/package.json#L57) await is implicit:

https://webdriver.io/docs/setuptypes.html#the-wdio-testrunner

It can still be something we want to enforce anyway

Reply:
Yeah, I'm not sure their documentation is correct since all the commands still returns Promises  and I had some issues when not using `await` :neutral_face:.

But it should be enforced by the [`@typescript-eslint/no-floating-promises`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md) ESLint rule so we should not have any problem with this.